src/lib/core/base/base-adapter.ts

```diff
@@ -38,6 +39,20 @@ export class BaseAdapter<T extends IBaseComponent> implements IBaseAdapter {
     toggleAttribute(this._component, hasAttribute, name, value);
   }
 
+  public redispatchEvent(event: Event): boolean {
+    const isFromLightDom = !((event.target as HTMLElement)?.getRootNode() instanceof ShadowRoot);
```

Comment:
The name could be clearer but this shouldn't cause any errors because the next step checks whether the event is composed. In practice this should prevent cross-shadow root usage.